src/widgetastic/widget/checkbox.py

```diff
@@ -31,3 +31,8 @@ def fill(self, value):
                 # TODO: More verbose here
                 raise WidgetOperationFailed('Failed to set the checkbox to requested value.')
             return True
+
+    @property
+    def is_enabled(self):
+        # TODO: This property can be implement globally.
+        return self.browser.element(self).is_enabled()
```

Comment:
As the TODO says, this shouldn't be implemented here, but in `Widget` itself.

There's already a `Widget.__element__` method that resolves the widget element through the correct browser and locator, which should be used instead of `self.browser.element(self)`.  This 'private' method should likely have a 'public' property, which an implementation of `is_enabled` can then use.

Reply:
@mshriver @izapolsk I added with `__element__` also added warning that user can overwrite this property if locator not pointing to specific webelement.

Reply:
@digitronik, @mshriver, I just looked thru this logic and I guess I was mistaken about __element__. 
As I see we shouldn't call it directly. it is part of element()->elements()->_process_locator() resolution chain.
I lean towards saying that first way was correct and we need to call browser.element() instead.